### PR TITLE
FE-236 Typeform component

### PR DESCRIPTION
**Pull request by jbianciotto**
# Description

Add a new Typeform component in /packages to embed or display a typeform in a modal.
Add support for reading form and country id query params to obtain the id of the typeform form 

# Jira tickets

https://trialtech.atlassian.net/browse/FE-236
https://trialtech.atlassian.net/browse/FE-240

# Screenshots

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue) :bug:
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) :bomb::boom:
- [ ] Configuration / Setup :wrench:
- [ ] Clean up / Technical debt :hammer:

## Apps affected by this change

- [x] Web
- [ ] Landing
- [ ] Management portal

## Project code modified by this change

- [X] Web
- [ ] Landing
- [ ] Management portal
- [X] Packages

# Checklist:

- [X] I have performed a self-review of my own code :innocent:
- [X] My code is clean and it follows the guidelines of this project :nail_care:
- [X] My changes don't generate new warnings :ok_hand:

### CODE REVIEW

File: packages/ui/components/typeform/typeform.types.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { PropsWithChildren } from 'react';
+import { PropsWithClassName } from '../../types/core';
+
+export type TypeFormProps = PropsWithChildren &
+  PropsWithClassName & {
+    formId: string;
+    hidden?: any;
```
Comment (libattistella) [inline packages/ui/components/typeform/typeform.types.ts:7]:
```suggestion
    hidden?: {[key: string]: any};
```

File: apps/web/package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "@vercel/edge-config": "^1.0.0",
     "clsx": "^1.2.1",
     "next": "^13.4.1",
+    "query-string": "^9.0.0",
```
Comment (libattistella) [inline apps/web/package.json:15]: Remove this and the install it in `packages/utils` and add a `getQueryString` function to a new file `packages/utils/strings.ts`. Then use that function from utils in the web project

File: apps/web/app/[lang]/studies/[id]/helpers.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
-import { Study } from '@models/study.types';
+import { Form, Study } from '@models/study.types';
 import { Gender } from './constants';
 import { useTranslations } from 'intl';
+import queryString from 'query-string';
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/helpers.ts:4]: Remove, you should import the utils function here

**Review by libattistella — CHANGES_REQUESTED** [review]
Nice!

File: apps/web/app/[lang]/studies/[id]/helpers.ts [hunk 4]
```diff
@@ -53,3 +54,39 @@ export const GenderToText = (gender: string) => {
...
   }
 };
+
+export const getTypeformIds = (
+  study: Study,
+  campaignId: string,
+  countryId: string,
+) => {
+  let form: Form;
+
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/helpers.ts:64]: This is overriding the previous if statement which is suppose to have preference. 

Apply this change please, I already checked that works fine. Replace this code 

```
  if (campaignId) {
    form = study.forms.find((form) => form.id === campaignId);
  }
  if (countryId) {
    form = study.forms.find((form) => form.country_id === countryId);
  } else {
    return {};
  }
```
with
 ```
  if (!campaignId?.length && !countryId?.length) return {};

  if (campaignId) {
    form = study.forms.find((form) => form.id === campaignId);
  } else {
    form = study.forms.find((form) => form.country_id === countryId);
  }
```

File: apps/web/app/[lang]/studies/[id]/helpers.ts [hunk 5]
```diff
@@ -53,3 +54,39 @@ export const GenderToText = (gender: string) => {
...
+  if (campaignId) {
+    form = study.forms.find((form) => form.id === campaignId);
+  }
+  if (countryId) {
+    form = study.forms.find((form) => form.country_id === countryId);
+  } else {
+    return {};
+  }
+
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/helpers.ts:73]: We are not checking if the form was found actually. We must check this. Apply this suggestion
```suggestion
if (!form) return {};
```

File: apps/web/app/[lang]/studies/[id]/helpers.ts [hunk 6]
```diff
@@ -53,3 +54,39 @@ export const GenderToText = (gender: string) => {
...
+    parsedCampaignId = parsedCampaignId[0];
+  }
+
+  let label = typeformParams.label;
+  if (Array.isArray(label)) {
+    label = label[0];
+  }
+
+  return {
+    typeformId: typeformId,
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/helpers.ts:88]:
```suggestion
    typeformId,
```

File: apps/web/app/[lang]/studies/[id]/helpers.ts [hunk 7]
```diff
@@ -53,3 +54,39 @@ export const GenderToText = (gender: string) => {
...
+    campaignId: parsedCampaignId,
+    label: label,
```
Comment (libattistella) [inline apps/web/app/[lang]/studies/[id]/helpers.ts:90]:
```suggestion
    label,
```

**Review by libattistella — CHANGES_REQUESTED** [review]
The component is not responding in mobile. It can't be closed neither start the journey

**Comment by jbianciotto** [comment]
> The component is not responding in mobile. It can't be closed neither start the journey
This may be an issue with Chrome developer tools.
I tested on physical devices ( chromie in iPhoneX, chrome in Redmi Note 8, MI browser in Redmi Note 8)  and had no issues with the typeform

**Review by libattistella — APPROVED** [review]

**Pull request merged**